### PR TITLE
feat: sync scroll following interactions in timegrid

### PR DESCRIPTION
**Pull request by adhrinae**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

- Implemented scroll syncing when the user performs certain interactions like event moving, event resizing, selecting grid in timegrid. 

https://user-images.githubusercontent.com/14539203/158553727-f62d4b62-d199-4266-b423-dfd6820d4b71.mov

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/calendar/src/hooks/timeGrid/timeGridScrollSync.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ) {
+      const { offsetTop, offsetHeight, scrollHeight } = scrollArea;
+      // Set minimum scroll boundary to the height of one row.
+      const scrollBoundary = Math.floor(scrollHeight / rowCount);
+      const layoutHeight = offsetTop + offsetHeight;
+      // arbitrary value to move scroll (px).
+      const SCROLL_OFFSET = 10;
+
+      if (y < offsetTop + scrollBoundary) {
+        scrollArea.scrollBy(0, -SCROLL_OFFSET);
```
Comment (adhrinae) [inline apps/calendar/src/hooks/timeGrid/timeGridScrollSync.ts:28]: IE는 ScrollBy를 지원하지 않음

File: apps/calendar/src/hooks/timeGrid/timeGridScrollSync.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      // Set minimum scroll boundary to the height of one row.
+      const scrollBoundary = Math.floor(scrollHeight / rowCount);
+      const layoutHeight = offsetTop + offsetHeight;
+
+      let scrollDiff: number;
+      if (y < offsetTop + scrollBoundary) {
+        scrollDiff = y - (offsetTop + scrollBoundary);
+        scrollArea.scrollTop = Math.max(0, scrollArea.scrollTop + scrollDiff);
+      } else if (y > layoutHeight - scrollBoundary) {
+        scrollDiff = y - (layoutHeight - scrollBoundary);
```
Comment (lja1018) [inline apps/calendar/src/hooks/timeGrid/timeGridScrollSync.ts:30]: `scrollDiff`를 if-else 문 안에서 선언해서 사용하는건 어떠신가요?

**Review by lja1018 — APPROVED** [review]
리뷰 완료합니다.

File: apps/calendar/src/hooks/timeGrid/timeGridScrollSync.ts (hunk 2 above)
Comment (adhrinae) [reply apps/calendar/src/hooks/timeGrid/timeGridScrollSync.ts:30]: 원래 처음에는 각각의 블록 안에서 `scrollDiff` 를 선언했었다가 정 반대의 코멘트가 달릴거라 생각해서 변수를 끌어올렸던건데 의외네요 ㅎㅎ 저도 처음에 각각 선언되는게 더 낫다고 생각했었으니 바꿀게요.

**Comment by adhrinae** [comment]
파이어폭스만 부정확하게 실패하고 있는데 로컬에서는 통과하는 것으로 판단되어 일단 머지하겠습니다.

**Pull request merged**